http_api.go

```diff
@@ -0,0 +1,194 @@
+			return
+		}
+
+		// pass it on!
+		next.ServeHTTP(w, req)
+	})
+}
+
+// HealthCheck indicates whether worker is currently functioning in a healthy
+// way. This can be used by a system like Kubernetes to determine whether to
+// replace an instance of worker with a new one.
+func (api *APIHandler) HealthCheck(w http.ResponseWriter, req *http.Request) {
+	// TODO actually check that processors are running and ready
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprint(w, "OK")
+}
+
+// GetWorkerInfo writes a JSON payload with useful information about the current
+// state of worker as a whole.
+func (api *APIHandler) GetWorkerInfo(w http.ResponseWriter, req *http.Request) {
+	info := api.i.workerInfo()
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(info)
+}
+
+// UpdateWorkerInfo allows reconfiguring some parts of worker on the fly.
+//
+// The main use of this is adjusting the size of the processor pool without
+// interrupting existing running jobs.
+func (api *APIHandler) UpdateWorkerInfo(w http.ResponseWriter, req *http.Request) {
+	var info workerInfo
+	if err := json.NewDecoder(req.Body).Decode(&info); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(errorResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	if info.PoolSize > 0 {
+		api.i.ProcessorPool.SetSize(info.PoolSize)
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// ShutdownWorker tells the worker to shutdown.
+//
+// Options can be passed in the body that determine whether the shutdown is
+// done gracefully or not.
+func (api *APIHandler) ShutdownWorker(w http.ResponseWriter, req *http.Request) {
+	var options shutdownOptions
+	if err := json.NewDecoder(req.Body).Decode(&options); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(errorResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	if options.Graceful {
+		api.i.ProcessorPool.GracefulShutdown(options.Pause)
+	} else {
+		api.i.cancel()
+	}
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// IncrementPool tells the worker to spin up another processor.
+func (api *APIHandler) IncrementPool(w http.ResponseWriter, req *http.Request) {
+	api.i.ProcessorPool.Incr()
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// DecrementPool tells the worker to gracefully shutdown a processor.
+func (api *APIHandler) DecrementPool(w http.ResponseWriter, req *http.Request) {
+	api.i.ProcessorPool.Decr()
+	w.WriteHeader(http.StatusNoContent)
+}
+
+type workerInfo struct {
+	Version          string `json:"version"`
+	Revision         string `json:"revision"`
+	Generated        string `json:"generated"`
+	Uptime           string `json:"uptime"`
+	PoolSize         int    `json:"poolSize"`
+	ExpectedPoolSize int    `json:"expectedPoolSize"`
+	TotalProcessed   int    `json:"totalProcessed"`
+
+	Processors []processorInfo `json:"processors"`
+}
+
+func (i *CLI) workerInfo() workerInfo {
```

Comment:
I would like these methods to live alongside their respective struct definitions :heart: